meshtastic/interfaces/ble/receive_compat_service.py

```
@@ -0,0 +1,662 @@
+        if cached_controller is not None:
+            return cached_controller
+        controller = controller_cls(iface)
+        state_lock = getattr(iface, "_state_lock", None)
+        if state_lock is None or _is_unconfigured_mock_member(state_lock):
+            with contextlib.suppress(
+                Exception
+            ):  # noqa: BLE001 - best-effort cache attach
+                iface._receive_recovery_controller = controller
+            return controller
+        with contextlib.suppress(Exception):  # noqa: BLE001 - best-effort cache attach
+            with state_lock:
```

Ruff (0.15.6)

[warning] Use a single `with` statement with multiple contexts instead of nested `with` statements

(SIM117)